src/types/varnum.rs

```diff
@@ -0,0 +1,195 @@
+//! Protocol Buffer Varints.
+
+use std::old_io::IoResult;
+use std::iter::range_step;
+
+use packet::Protocol;
+
+/// Protocol Buffer Varint, encoding a two's complement signed 32-bit integer.
+#[derive(Copy, Debug)]
+pub struct Varint;
```

Comment:
Using `varint` requires attribute `#[allow(non_camel_case_types)]`, that's why I went for `Varint` (it was `vi32`).